fix for PTV-1133 / PUBLIC-491 (read 'functions' field in features for Genome-9.0)

dcchivian opened the pull request:
passing unit tests

JamesJeffryes commented on lib/kb_phylogenomics/kb_phylogenomicsImpl.py:
```diff
@@ -3375,7 +3395,9 @@ def view_fxn_profile_phylo(self, ctx, params):
                         #    self.log (console, "prot: "+str(feature['protein_translation']))  # DEBUG
 
-                        if 'function' in feature and feature['function'] != None and feature['function'] != '':
+                        if 'function' in feature and feature['function'] != None and feature['function'] != '' \
```
You can write this line more concisely as `if feature.get('function') or feature.get('functions')`.

`.get()` returns `None` for missing keys (or provide a second argument to supply the default value) and `None`, `""`, and `[]` are falsey

JamesJeffryes commented on lib/kb_phylogenomics/kb_phylogenomicsImpl.py:
```diff
@@ -3392,7 +3414,16 @@ def view_fxn_profile_phylo(self, ctx, params):
+                                    raise ValueError ("need either function or functions for gene " \
+                                                      +gene_name+" in genome object "+str(genome_ref) \
+                                                      +" genome "+genome_sci_name_by_ref[genome_ref])
+
```
see above

dcchivian replied:
fixed

JamesJeffryes commented on lib/kb_phylogenomics/kb_phylogenomicsImpl.py:
```diff
@@ -2229,7 +2240,16 @@ def view_fxn_profile_featureSet(self, ctx, params):
+                                    raise ValueError ("need either function or functions for gene " \
+                                                      +gene_name+" in genome object "+str(genome_ref) \
+                                                      +" genome "+genome_sci_name_by_ref[genome_ref])
+
```
see above

JamesJeffryes commented on lib/kb_phylogenomics/kb_phylogenomicsImpl.py:
```diff
@@ -2209,7 +2219,8 @@ def view_fxn_profile_featureSet(self, ctx, params):
 
-                        if 'function' in feature and feature['function'] != None and feature['function'] != '':
+                        if ('function' in feature and feature['function'] != None and feature['function'] != '') \
+                           or ('functions' in feature and feature['functions'] != None and len(feature['functions']) != 0):
```
see above

JamesJeffryes commented on lib/kb_phylogenomics/kb_phylogenomicsImpl.py:
```diff
@@ -1060,7 +1060,8 @@ def view_fxn_profile(self, ctx, params):
 
-                        if 'function' in feature and feature['function'] != None and feature['function'] != '':
+                        if ('function' in feature and feature['function'] != None and feature['function'] != '') \
+                           or ('functions' in feature and feature['functions'] != None and len(feature['functions']) != 0):
```
You can write this line more concisely as if feature.get('function') or feature.get('functions').

.get() returns None for missing keys (or provide a second argument to supply the default value) and None, "", and [] are falsey

JamesJeffryes commented on lib/kb_phylogenomics/kb_phylogenomicsImpl.py:
```diff
@@ -1077,7 +1078,16 @@ def view_fxn_profile(self, ctx, params):
+                                    raise ValueError ("need either function or functions for gene " \
+                                                      +gene_name+" in genome object "+str(genome_ref) \
+                                                      +" genome "+genome_sci_name_by_ref[genome_ref])
+                                                      
```
You have already checked for this above so no need for the value error

JamesJeffryes commented on RELEASE_NOTES.md:
```diff
@@ -1,10 +1,13 @@
+### Version 1.2.2
+__Changes__
+- read 'functions' field in view functional profile apps (in KBaseGenomes.Genome-9.0 data typedef)
```
Should also mention removal of `trim_tree_to_genomeSet()`

dcchivian replied:
added

sychan commented on kbase.yml:
```diff
@@ -8,7 +8,7 @@ service-language:
+    1.2.2
 
 owners:
     [dylan]
```
Looks like James also did some work on this, should we add him in this list as well?

JamesJeffryes replied:
I'm a catalog admin so I don't really need to be added to the owners unless you want it done for bookkeeping

sychan commented:
@dcchivian A general observation is that this legacy code is not at all flake8 compliant, and generates 100 warnings/errors in the Impl file. You can just run this reformatter on the code and then it should be in line with the coding standards: https://github.com/hhatto/autopep8

sychan commented:
Here is an example tox.ini file that you can drop into the root of the project that turns off the most annoying warnings, and only enables the one we care about:
[flake8]
max-line-length = 120
exclude =
  lib/biokbase,
  submodules,
  */prepare_deploy_cfg.py,
  */NarrativeRunner_server_test.py,
  test_scripts
  temp_test*
putty-ignore =
    */__init__.py : F401,E126
    *Impl.py : E265,E266
    *Server.py : E265,E266
    *Client.py : E265,E266

sychan commented:
There are actually a bunch of substantive errors/complaints that you may want to look into: variables declared but never used, and complaints about the strings you have in the regex.

dcchivian commented:
I'm not sure what the process is / should be for style maintenance.  I agree that this code should conform to the style you want, and have lint cleaned up, etc.  but that level of effort for revising what was supposed to be a set of functioning prototype Apps that implementation team would bring into their fold is more than I can dedicate right now.  @sychan can you help get this repo to conform?  I'll try to follow any style going forward now that it's been established.

dcchivian commented:
I'm just trying to fix an error that resulted in a public ticket.  There is lots of cleanup to do, but I think that should follow a plan and not delay fixes that a customer is waiting on going out the door.  It's your call.  Let me know how you want to handle cleanup in general.  There's lots to do, certainly on modules I wrote :-)

dcchivian commented:
@sychan I'm trying to be a good citizen and run autopep8, but it's not installed on docker01 and I don't have perms to install it.  Can you or @scanon add it please?

dcchivian commented:
ran autopep8 on kb_phylogenomicsImpl.py.  addressed all review points except for lint and other similar issues (@sychan, what process are you running to check for those issues?).  passing unit tests.  @sychan @JamesJeffryes please take another look.  Thanks!

sychan commented:
If James is good with the code, we can merge it

dcchivian commented:
Enabling Travis-CI
